File: src/06-challenge/solution.ts
export default function inBox(box: string[]): boolean {
	const regex: RegExp = new RegExp(/^# *\* *#$/g)

	let i: number
	let line: string

	for (i = 1; i < box.length - 1; i++) {
		line = box[i]
		if (regex.test(line)) return true
	}

	return false
}

File: src/06-challenge/solution.py
import re


def in_box(box: list[str]) -> bool:
    regex: re.Pattern[str] = re.compile(pattern=r"^# *\* *#$")

    for i in range(1, len(box) - 1):
        line: str = box[i]
        if regex.search(string=line) is not None:
            return True

    return False
